clippy_lints/src/semicolon_if_nothing_returned.rs

```diff
@@ -0,0 +1,67 @@
+use crate::utils::{in_macro, span_lint_and_sugg, sugg};
+use if_chain::if_chain;
+use rustc_errors::Applicability;
+use rustc_hir::{Block, ExprKind};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+
+declare_clippy_lint! {
+    /// **What it does:** Looks for blocks of expressions and fires if the last expression returns `()`
+    /// but is not followed by a semicolon.
+    ///
+    /// **Why is this bad?** The semicolon might be optional but when
+    /// extending the block with new code, it doesn't require a change in previous last line.
+    /// It's also more idiomatic.
```

Comment:
Is it? Is this mentioned in the style guide? If so, please add a link to the documentation.

Reply:
Oke tbh I did not find anything here https://doc.rust-lang.org/1.0.0/style/style/README.html.